src/main/java/com/cloudbees/jenkins/plugins/advisor/client/AdvisorClient.java

```diff
@@ -3,90 +3,117 @@
 import com.cloudbees.jenkins.plugins.advisor.client.model.AccountCredentials;
 import com.cloudbees.jenkins.plugins.advisor.client.model.ClientUploadRequest;
 import com.google.gson.Gson;
+import com.ning.http.client.*;
+import com.ning.http.multipart.FilePart;
+import jenkins.plugins.asynchttpclient.AHCUtils;
 import org.apache.commons.lang.StringUtils;
-import org.asynchttpclient.*;
-import org.asynchttpclient.proxy.ProxyServer;
-import org.asynchttpclient.request.body.multipart.FilePart;
 
-import java.util.concurrent.CompletableFuture;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.logging.Logger;
 
 public class AdvisorClient {
 
+  private static final Logger LOG = Logger.getLogger(AdvisorClient.class.getName());
+
   private final AsyncHttpClient httpClient;
   private final AccountCredentials credentials;
   private final Gson gson;
 
   public AdvisorClient(AccountCredentials accountCredentials) {
-    this.httpClient = new DefaultAsyncHttpClient();
+    this.httpClient = new AsyncHttpClient((
+        new AsyncHttpClientConfig.Builder()
+            .setRequestTimeoutInMs(AdvisorClientConfig.insightsUploadTimeoutMilliseconds())
+            .setProxyServer(AHCUtils.getProxyServer())
+            .setFollowRedirects(true)
+            .build()));
     this.credentials = accountCredentials;
     this.gson = new Gson();
   }
 
-  public CompletableFuture<String> doAuthenticate() {
-    BoundRequestBuilder requestBuilder = httpClient.preparePost(AdvisorClientConfig.loginURI())
-      .setHeader("Content-Type", "application/json")
-      .setBody(gson.toJson(credentials))
-      .setFollowRedirect(true);
-
-    addProxySettings(requestBuilder);
-
-    return requestBuilder.execute()
-      .toCompletableFuture()
-      .exceptionally(e -> {throw new InsightsAuthenticationException("Unable to authenticate. Reason: " + e.getMessage());})
-      .thenApply(this::getBearerToken);
-  }
-
-  private void addProxySettings(BoundRequestBuilder requestBuilder) {
-    if (StringUtils.isNotBlank(credentials.getProxyHost())) {
-      ProxyServer.Builder proxyBuilder = new ProxyServer.Builder(credentials.getProxyHost(), credentials.getProxyPort());
-      proxyBuilder.setNonProxyHosts(credentials.getNonProxyHosts());
-
-      if (StringUtils.isNotBlank(credentials.getProxyUsername())) {
-        Realm realm = new Realm.Builder(credentials.getProxyUsername(), credentials.getProxyPassword()).setScheme(Realm.AuthScheme.BASIC).build();
-        proxyBuilder.setRealm(realm);
-      }
-      requestBuilder.setProxyServer(proxyBuilder);
+  public ListenableFuture<String> doAuthenticate() {
+    try {
+      return httpClient.preparePost(AdvisorClientConfig.loginURI())
```

Comment:
The proxy piece is handled by the Jenkins httpclient piece?

Reply:
Yep - https://wiki.jenkins.io/display/JENKINS/Async+Http+Client+Plugin